datafusion/sql/src/planner.rs

```diff
@@ -224,7 +224,24 @@ impl PlannerContext {
     }
 }
 
-/// SQL query planner
+/// SQL query planner and binder
+///
+/// This struct is used to convert a SQL AST into a [`LogicalPlan`].
+///
+/// You can control the behavior of the planner by providing [`ParserOptions`].
+///
+/// It performs the following tasks:
+///
+/// 1. Name and type resolution (called "binding" in other systems). This
+///    phase looks up table and column names using the [`ContextProvider`].
+/// 2. Mechanical translation of the AST into a [`LogicalPlan`].
+///
+/// It does not perform type checking, semantic analysis, type coercion, or
```

Comment:
I'm not sure if [check_unnest_arg](https://github.com/apache/datafusion/blob/1626c00a1b2f14057c8c4357ef9ea349350bbb60/datafusion/sql/src/expr/function.rs#L450C19-L450C35) conflicts with the description of type checking.

Reply:
This is a good point that some non trivial amount of semantic analysis and checking is done in the planner. I removed this text to make it:

```rust
/// It does not perform type coercion, or perform optimization, which are done
/// by subsequent passes.
```